services/modules/helm-charts/helm-charts.tf

```diff
@@ -273,42 +479,75 @@ resource "helm_release" "vault" {
   depends_on = [helm_release.guardian-message-broker]
 }
 
+data "archive_file" "guardian-extensions" {
+  type        = "zip"
+  source_dir  = "${path.root}/modules/helm-charts/charts/guardian-extensions"
+  output_path = "/tmp/guardian-extensions.zip"
+}
+
+resource "helm_release" "ingress-nginx" {
+  count      = var.use_ingress ? 1 : 0
+  name       = "ingress-nginx"
+  chart      = "ingress-nginx"
+  version    = "4.1.4"
+  repository = "https://kubernetes.github.io/ingress-nginx"
+}
+
 resource "helm_release" "guardian-extensions" {
-  name  = "guardian-extensions"
-  chart = "${path.root}/modules/helm-charts/charts/guardian-extensions"
+  name = "guardian-extensions"
 
-  timeout = "500"
+  chart = "${path.root}/modules/helm-charts/charts/guardian-extensions"
 
+  timeout      = "100"
+  force_update = true
   values = [
     "${file("${path.root}/modules/helm-charts/charts/guardian-extensions/values.yaml")}"
   ]
-}
 
-locals {
-  whitelist = join(",", var.whitelisted_ips)
-}
+  set {
+    name  = "global.guardian.whitelistedIps"
+    value = "{${local.ingress_whitelisted_ips}}"
+  }
 
-resource "helm_release" "extensions" {
-  for_each = toset(var.custom_helm_charts)
-  name     = each.value
-  chart    = each.value
-  repository = var.custom_helm_repository
+  set {
+    name  = "global.enable_newrelic"
+    value = var.enabled_newrelic
+  }
+  set {
+    name  = "global.newrelic_license_key"
+    value = var.newrelic_license_key
+  }
 
-  repository_username = var.custom_helm_repository_username
-  repository_password = var.custom_helm_repository_password
+  set {
+    name  = "global.apm_labels"
+    value = "env:${var.stage}"
+  }
 
-  values = [
-    "${file("${path.root}/custom_values.yaml")}"
-  ]
-#  set {
-#    name  = "global.guardian.whitelistedIps"
-#    value = "{${local.whitelist}}"
-#  }
   set {
-    name  = "global.SystemSchema"
-    value = var.system_schema
+    name  = "chart-sha1"
+    value = data.archive_file.guardian-extensions.output_sha
   }
+}
 
 
-  depends_on = [helm_release.guardian-message-broker]
-}
\ No newline at end of file
+# resource "helm_release" "extensions" {

```

Comment:
We will need to get this working so we can apply external charts to the stack too but this can be left for a later release if you have incorporated everything we need right now :)

Reply:
This is the next works, I will works on